utils/crypto/ledger/ledger.go

```diff
@@ -40,21 +42,36 @@ func addressPath(index uint32) string {
 }
 
 func (l *Ledger) Address(hrp string, addressIndex uint32) (ids.ShortID, error) {
-	_, hash, err := l.device.GetPubKey(addressPath(addressIndex), true, hrp, "")
+	resp, err := l.device.GetPubKey(addressPath(addressIndex), true, hrp, "")
 	if err != nil {
 		return ids.ShortEmpty, err
 	}
-	return ids.ToShortID(hash)
+	return ids.ToShortID(resp.Hash)
 }
 
 func (l *Ledger) Addresses(addressIndices []uint32) ([]ids.ShortID, error) {
+	if l.epk == nil {
+		pk, chainCode, err := l.device.GetExtPubKey(rootPath, false, "", "")
+		if err != nil {
+			return nil, err
+		}
+		l.epk = &bip32.Key{
+			Key:       pk,
+			ChainCode: chainCode,
+		}
+	}
+	// get first derivation of 0/v path
+	child0, err := l.epk.NewChildKey(0)
+	if err != nil {
+		return nil, err
+	}
 	addresses := make([]ids.ShortID, len(addressIndices))
-	for i, v := range addressIndices {
-		_, hash, err := l.device.GetPubKey(addressPath(v), false, "", "")
+	for i, child1Index := range addressIndices {
+		child1, err := child0.NewChildKey(child1Index)
```

Comment:
Pretty nit-picky, but I find the naming `child0` and `child1` pretty confusing. Mainly because they would seemingly be derived from the same parent `l.epk` - but instead `child1` is actually being derived from `child0`. Are there more standard names for these?